### ISSUE TITLE
Cannot connect to device on Windows (Trac #32)

### ISSUE THREAD

**Issue by cfobel** [wheeler-microfluidics/microdrop#36]
OS: Windows 7
System: Dell Latitude D620

-device plugged in and recognized in Windows device manager as COM4
-when running microdrop, the following warning is displayed:

Warning: no control board connected.

Migrated from http://microfluidics.utoronto.ca/ticket/32
```json
{
    "status": "closed", 
    "changetime": "2014-04-17T19:39:01", 
    "description": "OS: Windows 7\nSystem: Dell Latitude D620\n\n-device plugged in and recognized in Windows device manager as COM4\n-when running microdrop, the following warning is displayed:\n\nWarning: no control board connected.", 
    "reporter": "cfobel", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1397763541728826", 
    "component": "microdrop", 
    "summary": "Cannot connect to device on Windows", 
    "priority": "major", 
    "keywords": "", 
    "version": "0.1", 
    "time": "2012-01-05T19:56:31", 
    "milestone": "Microdrop 1.0", 
    "owner": "cfobel", 
    "type": "defect"
}
```

**Comment by cfobel** [comment]
Trac update at 20120107T01:18:36: 
* christian@fobel.net commented:

> commit 7049a33a248318f1f7c2900a0c161fbaaa1a1238
> Author: Christian Fobel <christian@fobel.net>
> Date:   Fri Jan 6 20:10:34 2012 -0500
> 
>     [SUBMODULE] Update dmf_control_board plugin...
>     
>     Give better feedback for connection problem on Windows.  Connection to dmf
>     device fails if there is a version mismatch between the device and the plugin.
>     Show a warning dialog to specify the version mismatch as the reason for
>     connection failure. Fixes #36.


* christian@fobel.net changed resolution from "" to "fixed"
* christian@fobel.net changed status from "new" to "closed"

**Comment by cfobel** [comment]
Trac update at 20120107T01:20:58: cfobel commented:

> Replying to [comment:1 christian@…]:
> > commit 7049a33a248318f1f7c2900a0c161fbaaa1a1238
> > Author: Christian Fobel <christian@fobel.net>
> > Date:   Fri Jan 6 20:10:34 2012 -0500
> > 
> >     [SUBMODULE] Update dmf_control_board plugin...
> >     
> >     Give better feedback for connection problem on Windows.  Connection to dmf
> >     device fails if there is a version mismatch between the device and the plugin.
> >     Show a warning dialog to specify the version mismatch as the reason for
> >     connection failure. Fixes #36.
> 
> This fix also enumerates the COM ports available in Windows and only tries to connect to those ports (rather than brute-forcing through ports 0-31).  Also, a delay of 0.1 seconds has been added between connection attempts to prevent spurious connection failures even when the device is connected.

**Comment by cfobel** [comment]
Trac update at 20140417T19:39:01: ryan changed component from "core" to "microdrop"
